cometbft/cometbft
# configure the gci linter

- remove unused linters that produced a warning and ensure that unparam is enabled
- apply the linters to tests
- additional autofixes
- test helpers
- all helper functions start from t.Helper()
- mainly assert -> require and other hygene changes
- further updates and attempts at fixes
- further cleanups
- unused-recievers
- mainly unused recievers
- Fix import paths and receiver names in various files
- Fix various return statements in the code
- "save game"
- fix double declaration
- b.helper()
- use stdlib http methods
- Fix test assertions in various test files
- Fix validation test assertions
- Fix test assertions and clean up code
- Fix field element comparison in TestGenPrivKeySecp256k1
- Fix various test failures and errors in the code
- Fix assertion in TestClientReplacesPrimaryWithWitnessIfPrimaryIsUnavailable
- Fix error handling and add arguments to execAnsible function
- as a piece of Free software, cometbft does NOT need to support digitalocean.
- start over with internal
- comment-spacings
- redoing internal
- Revert "b.helper()"
- error handling
- reactor_test.go removed in master
- fix a bug from removing an error that is always nil
- use correct require calls
- begin reversing expected and equal values
- complete reversing expected and actual
- favor require.NoError over require.Nil
- fix issue with len in kvstore test
- use the correct numeral type in TestBlockStoreSaveLoadBlock
- Fix test assertions in rpc/jsonrpc/client/args_test.go, mempool/clist_mempool_test.go, crypto/merkle/proof_test.go, and internal/store/store_test.go
- Fix assertions in test cases
- Fix test assertions in rpc/client/helpers_test.go and types/block_test.go
- revert a store test fix that was causing issues
- Fix test assertions in state_test.go, cache_test.go, clist_mempool_test.go, helpers_test.go, and status_test.go
- assert.Gerater
- Fix assert statements in test files
- Fix assertion order in TestBlockFetchAtHeight
- fix regressions
- Fix test assertions in various test files
- Fix test assertions in detector_test.go, netaddress_test.go, addrbook_test.go, and switch_test.go
- Add gofmt linter and update assertions in tests
- Fix test helper functions
- Refactor code to use strconv.Itoa instead of fmt.Sprintf
- Fix assertion errors in unit tests
- Fix test assertions in types_test.go, secp256k1_internal_test.go, group_test.go, pool_test.go, verify_test.go, and state_test.go
- Fix test cases and update provider configuration
- Fix ValidateBasic assertion in http_test.go
- attempt to fix a regression
- Refactor assert functions in verify_test.go
- remove gocritic linter and update assertions in tests
- Fix error handling in state_test.go and validator_set_test.go
- Fix test assertions in store_test.go, http_test.go, and validation_test.go
- Fix test assertions in unit tests
- Fix test assertions in unit tests
- Fix test assertions in store_test.go, setup_test.go, http_test.go, netaddress_test.go, rpc_test.go, http_json_handler_test.go, and tx_test.go
- attempt to fix http test
- Fix test assertions in peer_test.go, event_test.go, status_test.go, rpc_test.go, http_json_handler_test.go, parse_test.go, and results_test.go
- Fix various test assertions
- Fix test assertions in state_test.go, chunks_test.go, store_test.go, clist_mempool_test.go, peer_set_test.go, addrbook.go, switch_test.go, rpc_test.go, and validator_set_test.go
- Fix test assertions and remove unused import statements
- Fix test assertions in state_test.go and pruner_test.go
- Fix makeValidCommit function signature
- Fix assert statements and require.NoError usage in tests
- Fix test assertions and error handling
- Fix type conversion in TestSetValidBlockOnDelayedPrevote
- Fix test assertions in rpc/client/rpc_test.go and rpc/jsonrpc/server/parse_test.go
- Fix empty result assertion in TestTxSearch
- Fix variable initialization and remove unnecessary error handling
- remove unused code
- finished :D
- Fix reactorPairs initialization in TestBadBlockStopsPeer
- Fix empty result assertion in TestTxSearch
- Fix duplicate channel creation in TestNodeInfoValidate
- Fix typos and add comments to address book functionality
- Update field names in TestStruct
- Fix randGenesisDoc function signature and update function calls
- Add helper function for counting old and new addresses in selection
- Refactor calcOldBucket function in addrbook.go
- Fix code formatting and remove unused code
- Fix linter configuration
- test if this is a ci flake by adding some godoc
- fix lint issues in ci jobs
- add a godoc to fix tests
- add a godoc to see if another test is flaky
- Fix various code formatting issues
- add a godoc to determine if TestFromData is flaky
- Fix indexer method receiver names
- Refactor function signatures in switch_test.go and test_util.go
- Refactor BackportTxIndexer methods
- Fix method receivers in p2p/mock/peer.go
- Fix unimplemented methods in EventSink
- Refactor saveValidatorsInfo and saveConsensusParamsInfo functions
- Fix receiver parameter names in test files
- Fix variable naming and formatting issues in reactor_test.go, io_test.go, addrbook_test.go, pex_reactor.go, test_util.go, ws_client_test.go, jsonrpc_test.go, and http_json_handler_test.go
- revert changes to jsonrpc types, though I suspect this is more flake
- Fix code formatting and comments
- Remove unnecessary break statements
- Fix function return values and add missing return statements
- Fix variable naming and formatting issues in tests
- Fix variable naming and formatting issues in state_test.go, key.go, key_test.go, and addrbook_test.go
- Refactor code and improve test coverage
- Remove unnecessary code and fix import comments
- Refactor RPCError tests to use table-driven tests
- Fix JSON unmarshal error in types_test.go
- Fix import paths in JSON-RPC client and server files
- update btcutil
- revert a mistaken change
- Fix range iteration in ParseMetricsDir function
- Revert "Fix import paths in JSON-RPC client and server"
- cleanup
- finish with revive
- use gci to sort imports well



---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments
